py/sentry_streams/adapters/loader.py

```diff
@@ -1,14 +1,19 @@
 from enum import Enum
 from typing import Union, assert_never
 
-from sentry_streams.adapters.stream_adapter import PipelineConfig, StreamAdapter
+from sentry_streams.adapters.stream_adapter import (
+    PipelineConfig,
+)
+from sentry_streams.flink.flink_adapter import FlinkAdapter
 
 
 class AdapterType(Enum):
     FLINK = "flink"
 
 
-def load_adapter(adapter_type: Union[AdapterType, str], config: PipelineConfig) -> StreamAdapter:
+def load_adapter(
+    adapter_type: Union[AdapterType, str], config: PipelineConfig
+) -> Union[FlinkAdapter]:
```

Comment:
Also unfortunate. I can't get it to work unless I do this or return `StreamAdapter[...Flink specific types...]`.

Reply:
You can do this as a follow up